services/workflows-service/src/workflow/context-changed-webhook-caller.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { randomUUID } from 'crypto';
+
+export class ContextChangedWebookCaller {
+  #__webhookConfig: ProcessEnvWebhookConfig;
+  #__httpService;
+
+  constructor(eventEmitter: any, webhookConfig: any, httpService: any) {
+    this.#__webhookConfig = webhookConfig;
+    this.#__httpService = httpService;
+
+    eventEmitter.subscribe('workflow.context.changed', this.callWebhook.bind(this));
+  }
+
+  callWebhook(data: any) {
+    const { runtimeData, state, newContext, ..._ } = data;
+
+    const oldDocuments = runtimeData.context['documents'] || [];
+    const newDocuments = newContext['documents'] || [];
+
+    const anyDocumentStatusChanged = (oldDocuments as Array<any>).some(
+      (oldDocument, i) => oldDocument.decision.status !== newDocuments[i].decision.status,
+    );
+
+    const id = randomUUID();
+
+    setTimeout(() => {
+      this.#__httpService.post(this.#__webhookConfig.url, {
+        id,
```

Comment:
eventually, i think this should be extracted to a webhook service, he will hold the knowledge webhook data wrapper and other logic related to sending the webhook etc... - so we can handle additional events quickly (if a user asks to get also XYZ event) - and also can quickly extract it to another process (to improve reliability)

It's not an immediate need but I think its smart to have it so we can react quickly

Reply:
move to webhook service